### PR TITLE
Fix DLEQ proof verification

### PR DESCRIPTION
**Pull request by alxdavids**
- Proof computation was incorrect (Issue #2 )
-- We now explicitly recompute PRNG (SHAKE256) seed
-- Do not check unnecessary proof values sent by server
- Include browserify'd version of keccak npm module for SHAKE256 (compatible with challenge-bypass-server)
- Updated readme to be consistent with updated verification
- Updated manifest version to v1.0.1
- Tested receiving signed tokens in Chrome & Firefox successfully

(also more reviewers welcome)

### CODE REVIEW

File: scripts/crypto.js [hunk 1]
```diff
@@ -479,38 +419,31 @@ function getMarshaledBatchProof(proof) {
...
+
+// Return a bignum from a hex string
+function getBigNumFromHex(hex) {
+    return sjcl.bn.fromBits(sjcl.codec.hex.toBits(hex));
+}
+
+// PRNG encode point
+function encodePointForPRNG(point) {
+    let hex = sjcl.codec.hex.fromBits(point.toBits());
+    let newHex = "04" + hex;
```
Comment (lziest) [inline scripts/crypto.js]: is there anything special about this prefix?

File: scripts/crypto.js [hunk 2]
```diff
@@ -353,120 +339,74 @@ function verifyDleq(bp, chkM, chkZ) {
...
-        console.error(INCONSISTENT_DLEQ_PROOF_ERR);
-        return false;
-    }
-    return true;
+
+    return {M: cM, Z: cZ};
 }
 
-// Checks that the batch proof is well-defined
-function isBatchProofCompleteAndSane(bp, chkM, chkZ) {
```
Comment (lziest) [inline scripts/crypto.js]: Nice. This gist of this deletion is that we don't need to check the point formation since we just verify the proof, right?

Comment (alxdavids) [reply scripts/crypto.js]: Yes that's right. We used to send all the proof values and check they were the same after recomputation. Then we verified the proof using the values that were sent by the server. This is a suboptimal approach and not as the protocol is described, exactly.

We now just verify the proof directly using only the (c,s) values as detailed in the protocol.

File: scripts/crypto.js (hunk 1 above)
Comment (alxdavids) [reply scripts/crypto.js]: On the server-side all the points have this prefix. I'll make it static and add a comment explaining this though.

Comment (gtank) [reply scripts/crypto.js]: The byte 0x04 means "uncompressed" for SEC1 marshaling. 0x02 or 0x03 would indicate compression with a particular sign (as seen [here](https://github.com/privacypass/challenge-bypass-extension/blob/master/scripts/crypto.js#L214) and [here](https://github.com/privacypass/challenge-bypass-server/blob/master/crypto/curve.go#L65))

Comment (lziest) [reply scripts/crypto.js]: cool, assume this is the reference doc? http://www.secg.org/sec1-v2.pdf

**Comment by alxdavids** [comment]
I've made the point encoding clearer in the latest commit. Has anyone had chance to check that the DLEQ verification steps match the protocol spec?

**Comment by lziest** [comment]
@grittygrease I don't have privilege to merge this. :)

**Pull request merged**